client/app/bundles/course/assessment/submission/components/ScribingView/ScribingCanvas.jsx

```diff
@@ -0,0 +1,549 @@
+      );
+      this.canvas.setBackgroundImage(fabricImage, this.canvas.renderAll.bind(this.canvas));
+
+      this.canvas.initializeScribbles = this.initializeScribbles;
+      this.canvas.initializeScribbles(true);
+
+      this.canvas.on('mouse:down', this.onMouseDownCanvas);
+      this.canvas.on('mouse:move', this.onMouseMoveCanvas);
+      this.canvas.on('mouse:up', this.onMouseUpCanvas);
+      this.canvas.on('mouse:over', this.onMouseOver);
+      this.canvas.on('mouse:out', this.onMouseOut);
+      this.canvas.on('object:moving', this.onObjectMovingCanvas);
+      this.canvas.on('object:modified', this.saveScribbles);
+      this.canvas.on('object:added', this.saveScribbles);
+      this.canvas.on('path:created', this.saveScribbles);
+      this.canvas.on('object:removed', this.saveScribbles);
+      this.canvas.on('text:changed', this.saveScribbles);
+
+      this.scaleCanvas();
+      this.props.setCanvasLoaded(this.props.answerId, true, this.canvas);
+    };
+  }
+
+  // Adjusting canvas height after canvas initialization
+  // helps to scale/move scribbles accordingly
+  scaleCanvas() {
+    this.canvas.setWidth(this.width);
+    this.canvas.setHeight(this.height);
+    this.canvas.renderAll();
+  }
+
+  // Scribble Helpers
+
+  saveScribbles = () => (
+    new Promise((resolve) => {
+      if (this.isScribblesLoaded) {
+        const answerId = this.props.answerId;
+        const json = this.getScribbleJSON();
+        this.props.updateScribingAnswerInLocal(answerId, json);
+        this.props.updateScribingAnswer(answerId, json);
+      }
+      resolve();
+    })
+  )
+
+  getScribbleJSON() {
+    // Remove non-user scribings in canvas
+    this.canvas.layers.forEach((layer) => {
+      if (layer.creator_id !== this.props.scribing.answer.user_id) {
+        layer.showLayer(false);
+      }
+    });
+
+    // Only save rescaled user scribings
+    const objects = this.canvas._objects;
+    objects.forEach((obj) => {
+      this.normaliseScribble(obj);
+    });
+    const json = JSON.stringify(objects);
+
+    // Scale back user scribings
+    objects.forEach((obj) => {
+      this.denormaliseScribble(obj);
+    });
+
+    // Add back non-user scribings according canvas state
+    this.canvas.layers.forEach(layer => (layer.showLayer(layer.isDisplayed)));
+    return `{"objects": ${json}}`;
+  }
+
+  // Utility Helpers
+
+  getRgbaHelper = json => (
+    `rgba(${json.r},${json.g},${json.b},${json.a})`
+  );
+
+  getMousePoint = event => (
+    {
+      x: event.clientX,
+      y: event.clientY,
+    }
+  );
+
+  // Generates the left, top, width and height of the drag
+  generateMouseDragProperties = (point1, point2) => (
+    {
+      left: point1.x < point2.x ? point1.x : point2.x,
+      top: point1.y < point2.y ? point1.y : point2.y,
+      width: Math.abs(point1.x - point2.x),
+      height: Math.abs(point1.y - point2.y),
+    }
+  );
+
+  getCanvasPoint(event) {
+    const pointer = this.canvas.getPointer(event.e);
```

Comment:
Should this be `getPointer(event)`? 

`event.e` is `undefined` because you already did `.e` at the points where `getCanvasPoint` are called. Doesn't fail on Chrome because `getPointer` tries to use `fabric.window.event` instead, but this doesn't work on firefox.